mongodb/internal/configure-pipeline/execute-pipeline.py

```diff
@@ -0,0 +1,123 @@
+import kratix_sdk as ks
+import yaml
+
+def create_db(name, replicas, version):
+    """
+    create a request for a MongoDB database
+    """
+    mongo = {
+        "apiVersion": "mongodbcommunity.mongodb.com/v1",
+        "kind": "MongoDBCommunity",
+        "metadata": {
+            "name": name,
+            "namespace": "default"
+        },
+        "spec": {
+            "members": replicas,
+            "type": "ReplicaSet",
+            "version": version,
+            "security": {
+                "authentication": {
+                    "modes": ["SCRAM"]
+                }
+            },
+            "users": [
+                {
+                    "name": "my-user",
+                    "db": "admin",
+                    "passwordSecretRef": {
+                        "name": f"mongodb-password-{name}"
+                    },
+                    "roles": [
+                        {"name": "root", "db": "admin"},
+                        {"name": "dbAdminAnyDatabase", "db": "admin"},
+                        {"name": "clusterAdmin", "db": "admin"},
+                        {"name": "userAdminAnyDatabase", "db": "admin"},
+                    ],
+                    "scramCredentialsSecretName": "my-scram"
+                }
+            ],
+            "additionalMongodConfig": {
+                "storage.wiredTiger.engineConfig.journalCompressor": "zlib"
+            }
+        }
+    }
+    return mongo
+
+def create_secret(name, password):
+    """
+    Create a secret for the database
+    """
+    secret = {
+        "apiVersion": "v1",
+        "kind": "Secret",
+        "metadata": {
+            "name": name,
+            "namespace": "default"
+        },
+        "type": "Opaque",
+        "stringData": {
+            "password": password,
+        }
+    }
+
+    return secret
+
+def main():
+    # Initialise the sdk
+    sdk = ks.KratixSDK()
+
+    # Read the resource input
+    resource = sdk.read_resource_input()
+
+    # Get the name the resource
+    name = resource.get_name()
+
+    print("Creating mongodb with name:", name)
+    
+    # Set default values
+    replicas= 1
+    password= "supersecret"
+
+    # Get correct mongodb version
+    try:
+        major_version = resource.get_value("spec.majorVersion")
+    except:
+        print("major version not specified, defaulting to 6")
+        major_version = 6
```

Comment:
the (unreleased) version of the sdk supports the default value as a second argument!

Reply:
nice!! Will change after it's out :)